pkg/web/apps/apps_service.go

```diff
@@ -63,14 +63,25 @@ func (a *appsService) GetActiveAppByID(id string) (*models.App, error) {
 }
 
 // GetApps retrieves the list of apps from the repository
-func (a *appsService) UpdateAppVersions(versions []models.Version) error {
-	err := a.repository.UpdateAppVersions(versions)
+func (a *appsService) UpdateAppVersions(id string, versions []models.Version) error {
+
+	app, err := a.repository.GetActiveAppByID(id)
 
-	// Check for errors and return the appropriate error to the handler
 	if err != nil {
 		return err
 	}
 
+	for i := 0; i < len(versions); i++ {
+		if versions[i].AppID != app.AppID {
```

Comment:
What if 2/3 version app IDs are correct? Should we still update those?

Reply:
No. I don't think so. If it has any data inconsistent than the error should be returned. 
In this case, if any version informed in the list is not associated with the id of the app informed an error as Bad Request ( invalid id supplied ) will be returned. Is it make sense? @davidffrench do you agree with it?

Reply:
This type of request should never happen from our UI, so it is sufficient to send a Bad Request on the first incorrect id in my opinion.

Reply:
OK. The log was added. Could you please verify this one for we are able to move forward and merge it?

Reply:
done.